File: Singleton/singleton.py
# Python Singleton Pattern
class Singleton:
    _instance = None  # class attribute to store the single instance

    def __new__(cls, *args, **kwargs):        
        if not cls._instance: # If no instance exists, create one and assign it to _instance
            cls._instance = super(Singleton, cls).__new__(cls, *args, **kwargs)
        return cls._instance

# Example usage
singleton1 = Singleton()
singleton2 = Singleton()

print(singleton1 is singleton2)  # Output: True, both are the same instance

File: Singleton/singleton.ts
// TypeScript Singleton Pattern
class Singleton {
    private static instance: Singleton;
    private constructor() { // Private constructor prevents direct construction calls with 'new'
    }

    public static getInstance(): Singleton {
        if (!Singleton.instance) { // If no instance exists, create one
            Singleton.instance = new Singleton();
        }
        return Singleton.instance;
    }

    public someBusinessLogic() {
        console.log("Executing business logic");
    }
}

// Example usage of singleton
const instance1 = Singleton.getInstance();
const instance2 = Singleton.getInstance();
console.log(instance1 === instance2); // Output: True, both are the same instance
